ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/suites/ValueNetsingIT.scala

```diff
@@ -0,0 +1,116 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.testtool.suites
+
+import com.daml.ledger.api.refinements.ApiTypes.Party
+import com.daml.ledger.api.testtool.infrastructure.Allocation._
+import com.daml.ledger.api.testtool.infrastructure.Assertions._
+import com.daml.ledger.api.testtool.infrastructure.LedgerTestSuite
+import com.daml.ledger.api.testtool.infrastructure.participant.ParticipantTestContext
+import com.daml.ledger.test.semantic.ValueNesting._
+import io.grpc.Status
+
+import scala.annotation.tailrec
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Success
+
+final class ValueNestingIT extends LedgerTestSuite {
+
+  @tailrec
+  def toNat(i: Long, acc: Nat = Nat.Z(())): Nat =
+    if (i == 0) acc else toNat(i - 1, Nat.S(acc))
+
+  @tailrec
+  def toLong(n: Nat, acc: Long = 0): Int =
+    n match {
+      case Nat.Z(_) => 0
+      case Nat.S(n) => toLong(n, acc + 1)
+    }
+
+  def reify[X](future: Future[X])(implicit ec: ExecutionContext): Future[Either[Throwable, X]] =
```

Comment:
sounds like a bit of a weird name for this but I might just be unfamiliar with Scala conventions here.

Reply:
will change it.